src/js/utils/SchemaUtil.js

```diff
@@ -61,7 +61,7 @@ function schemaToFieldDefinition(fieldName, fieldProps, formParent, isRequired,
     definition.checked = value;
   }
 
-  if (fieldProps.fieldType === 'number') {
+  if (fieldProps.type === 'number') {
```

Comment:
I feel like this could have consequences elsewhere. We should check all of our schemas that are rendered into a SchemaForm to make sure they have the correct type set.